File: src/python/tutorias/domain/docente.py
from .usuario import Usuario


class Docente(Usuario):
    def __init__(self, id_usuario: int, nombre: str, correo: str, departamento: str):
        super().__init__(id_usuario, nombre, correo)
        self._departamento = departamento

    def publicar_horario(self):
        pass

    def consultar_horarios(self):
        pass

File: src/java/edu/uees/tutorias/domain/Docente.java
package edu.uees.tutorias.domain;

public class Docente extends Usuario {

    private String departamento;

    public Docente(
            int id,
            String nombre,
            String correo,
            String departamento) {

        super(id, nombre, correo);
        this.departamento = departamento;
    }

    public void publicarHorario() {
        // La gestión del horario se implementará con HorarioTutoria.
    }

    public void consultarHorarios() {
        // La consulta se coordinará con los horarios asociados al docente.
    }
}